internal/license/agent_test.go

```diff
@@ -98,38 +108,50 @@ func TestAgent(t *testing.T) {
 	})
 
 	t.Run("initial license retrieval fails and recovers", func(t *testing.T) {
-		upstreamClient := newMockKonnectLicenseClient(mo.None[license.KonnectLicense]())
+		ticker := mocks.NewTicker()
+
+		upstreamClient := newMockKonnectLicenseClient(mo.None[license.KonnectLicense](), ticker)
 
 		// Return an error on the first call to List() to verify that the agent handles this correctly.
 		upstreamClient.ReturnError(errors.New("something went wrong on a backend"))
 
 		const (
-			// Set the initial polling period to a very short duration to ensure that the agent retries quickly.
-			initialPollingPeriod = time.Millisecond
-			regularPollingPeriod = time.Millisecond * 5
-			allowedDelta         = time.Millisecond
+			initialPollingPeriod = time.Minute * 3
+			regularPollingPeriod = time.Minute * 20
+			allowedDelta         = time.Second
 		)
+
 		a := license.NewAgent(
 			upstreamClient,
 			logr.Discard(),
 			license.WithInitialPollingPeriod(initialPollingPeriod),
 			license.WithPollingPeriod(regularPollingPeriod),
+			license.WithTicker(ticker),
 		)
 
 		startTime := time.Now()
 		go func() {
 			err := a.Start(ctx)
 			require.NoError(t, err)
```

Comment:
`require` is not safe to call in created goroutines - read more https://github.com/stretchr/testify/pull/1392

Reply:
👍 The error returned from `Start()` isn't really all that useful since it will only return the error from the cancelled context so we can just fall back to `go a.Start(ctx)` here